compiler/ras/ILValidator.cpp

```diff
@@ -219,6 +220,14 @@ bool TR::ILValidator::treesAreValid(TR::TreeTop *start, TR::TreeTop *stop)
          if (isEndOfExtendedBlock)
             validateEndOfExtendedBlock(iter);
          }
+      else if (opcodeValue == TR::branch &&
+               iter.currentParent() != NULL &&
+               iter.currentParent()->getOpCodeValue() == tstart)
```

Comment:
I know the opcode properties table says that TR::branch has a single child, but I don't see any code anywhere that creates or treats a TR::branch node that has anything other than zero children.

So that makes me wonder if the TR::branch child property is wrong, as well as if this correctness condition is therefore more complicated than it needs to be. @magaudet @cathyzhyi can you shed light on TR::branch?

Reply:
The child property in the table is derived from a comment in `omr/compiler/il/OMRILOpCodesEnum.hpp:726`:

```c++
   branch,     // generic branch --> first child is a constant indicating the mask
```

If this is indeed incorrect, I'd be more than happy to simplify the validator and fix the child property (and the comment). Also, a follow up question from me (perhaps better palced in a separate issue), if TR::branch doesn't expect a child, what's the difference between TR::branch and TR::Goto?

Reply:
Botched Matthew's @ reference...@mgaudet do you happen to know the history of the TR::branch opcode ?

Reply:
@Leonardo2718 TR::Goto is a treetop, for one. TR::Case is a much more appropriate relationship.